### DISCUSSION TITLE
[Bug]:  Unaable to install 11.0.8 version.

### DISCUSSION THREAD

**Discussion by shree0** [pimcore/pimcore#15946] [General]
### Pimcore version

11.0.8

### Steps to reproduce

After this commnad `.--admin-username=xxx --admin-password=xxxx --mysql-username=xxxx --mysql-password=xxxx --mysql-database=xxxx --no-interaction` 
I am getting
`[ERROR] The following errors were encountered during installation `
`* Manage Indexes is required.` 
errors.
In `var/log/dev.log` I can see the following error message:
```
php.INFO: User Deprecated: Not configuring a schema manager factory is deprecated. Use Doctrine\DBAL\Schema\DefaultSchemaManagerFactory which is going to be the default in DBAL 4. (Connection.php:225 called by DriverManager.php:196, https://github.com/doctrine/dbal/issues/5812, package doctrine/dbal) {"exception":"[object] (ErrorException(code: 0): User Deprecated: Not configuring a schema manager factory is deprecated. Use Doctrine\\DBAL\\Schema\\DefaultSchemaManagerFactory which is going to be the default in DBAL 4. (Connection.php:225 called by DriverManager.php:196, https://github.com/doctrine/dbal/issues/5812, package doctrine/dbal) at /..../..../.../pimcore/vendor/doctrine/deprecations/lib/Doctrine/Deprecations/Deprecation.php:209)"} []
```
please help to resolve this.

### Actual Behavior

This commnad `.--admin-username=xxx --admin-password=xxxx --mysql-username=xxxx --mysql-password=xxxx --mysql-database=xxxx --no-interaction`  is not excuted.

### Expected Behavior

Not facing any issues during 10.x versions. But in 11.x getting error during installation.

**Comment by kingjia90** [comment]
Looks like the user is not allowed to manage indexes
See also https://github.com/pimcore/pimcore/blob/33173df4b983fe81035e44e8b7dadf956f49f5cc/doc/23_Installation_and_Upgrade/01_System_Requirements.md?plain=1#L64

**Comment by shree0** [comment]
But in the error log, I can see

 `php.INFO: User Deprecated: Not configuring a schema manager factory is deprecated. Use Doctrine\DBAL\Schema\DefaultSchemaManagerFactory which is going to be the default in DBAL 4. (Connection.php:225 called by DriverManager.php:196, https://github.com/doctrine/dbal/issues/5812, package doctrine/dbal) {"exception":"[object] (ErrorException(code: 0): User Deprecated: Not configuring a schema manager factory is deprecated. Use Doctrine\\DBAL\\Schema\\DefaultSchemaManagerFactory which is going to be the default in DBAL 4. (Connection.php:225 called by DriverManager.php:196, https://github.com/doctrine/dbal/issues/5812, package doctrine/dbal) at /..../..../.../pimcore/vendor/doctrine/deprecations/lib/Doctrine/Deprecations/Deprecation.php:209)"} []
`

error. I think this is causing the issue while installing.

**I am installing it in the CPanel server.**

**Comment by kingjia90** [comment]
No, deprecation warning are not stopping the exection or affecting it, these are just heads-up for future releases/upgrades, we are not even on DBAL4 for example, it's not affecting the code "yet" (otherwise it's a bc break and a potential regression from their end).

**Comment by shree0** [comment]
OK, Is there any other ways to trace my problem during installation?

**Comment by kingjia90** [comment]
the installation script works with multiple steps, right now is stopping because the user doesn't have the permission to manage the indexes and therefore it wouldn't be able to create/modify the table indexes and is stopping.
You just need to grant the permissions to the user (the same defined after `mysql-username=`) and should be good to go.

Any other errors, will interrupt the script from continuing within red blocks, can't miss/overlook them.

**Reply by shree0** [reply]
I still didn't get how to give permission to manage the indexes. I was trying to install it through the command line. ( I have tried by giving permission to the installation folder i.e.`chmod 777` )

Previously I was able to install 10.x series without these errors.

**Reply by kingjia90** [reply]
It's a SQL permission thing, not sure if you created a new user or so. Maybe locally you can try by using root

**Reply by shree0** [reply]
What is the path for this? OR Where database credentials are stored? (I will try to add my database credentials before installation)

We are installing this by using root in the CPanel server.

**Comment by shree0** [comment]
After giving the command `COMPOSER_MEMORY_LIMIT=-1 composer create-project pimcore/skeleton ./my-project` it was trying to install the 11.0.3 version. Is this the latest one?

What is download to install it by using the GET/POST method?

**Reply by kingjia90** [reply]
Yes, https://github.com/pimcore/skeleton/releases

**Reply by shree0** [reply]
I can't see the installation URL for the GET/POST method from this download link.

**Reply by kingjia90** [reply]
was replying to 
> the 11.0.3 version. Is this the latest one?

not sure what is the GET/POST method you mentioning , are you looking for these?
https://github.com/pimcore/skeleton.git
https://github.com/pimcore/skeleton/archive/refs/tags/v11.0.3.tar.gz

**Reply by shree0** [reply]
Accessing the installation URL through the browser.
Here in this guide, they mentioned installation through command (this method I am following to install)

**Comment by Sats-25** [comment]
Steps to reproduce
After this commnad .--admin-username=xxx --admin-password=xxxx --mysql-username=xxxx --mysql-password=xxxx --mysql-database=xxxx --no-interaction
I am getting
[ERROR] The following errors were encountered during installation 
* WITH RECURSIVE is required.
errors.
In var/log/dev.log I can see the following error message:

- [[2024-01-17T22:46:35.655161+05:30] php.INFO: User Deprecated: Not configuring a schema manager factory is deprecated. Use Doctrine\DBAL\Schema\DefaultSchemaManagerFactory which is going to be the default in DBAL 4. (Connection.php:226 called by DriverManager.php:197, https://github.com/doctrine/dbal/issues/5812, package doctrine/dbal) {"exception":"[object] (ErrorException(code: 0): User Deprecated: Not configuring a schema manager factory is deprecated. Use Doctrine\\DBAL\\Schema\\DefaultSchemaManagerFactory which is going to be the default in DBAL 4. (Connection.php:226 called by DriverManager.php:197, https://github.com/doctrine/dbal/issues/5812, package doctrine/dbal) at /var/www/html/demo11/vendor/doctrine/deprecations/lib/Doctrine/Deprecations/Deprecation.php:210)"} []
[2024-01-17T22:46:35.663641+05:30] console.DEBUG: Command "pimcore:install --admin-username admin --admin-password admin --mysql-host-socket='127.0.0.1:3310' --mysql-username root --mysql-password root --mysql-database demo_11 --no-interaction" exited with code "2" {"command":"pimcore:install --admin-username admin --admin-password admin --mysql-host-socket='127.0.0.1:3310' --mysql-username root --mysql-password root --mysql-database demo_11 --no-interaction","code":2} [] 
- ]
